src/phoenix/experimental/evals/evaluators.py

```diff
@@ -1,7 +1,107 @@
-from typing import List, Optional
+from dataclasses import dataclass
+from functools import partial
+from typing import Any, List, Mapping, Optional, Protocol
 
-from phoenix.experimental.evals import PromptTemplate
-from phoenix.experimental.evals.models import BaseEvalModel
+from phoenix.experimental.evals.models import set_verbosity
+from phoenix.utilities.logging import printif
+
+from .models import BaseEvalModel
+from .templates import ClassificationTemplate, PromptTemplate
+
+Record = Mapping[str, Any]
+
+NOT_PARSABLE = "NOT_PARSABLE"
+
+
+@dataclass(frozen=True)
+class EvaluationResult:
+    """
+    A class to contain the results of an evaluation.
+    """
+
+    prediction: str
+
+
+class Evaluator(Protocol):
+    """
+    Evaluates individual records.
+    """
+
+    def evaluate(self, record: Record) -> EvaluationResult:
+        ...
+
+    async def aevaluate(self, record: Record) -> EvaluationResult:
+        ...
+
+    @property
+    def name(self) -> str:
+        ...
+
+
+class LLMEvaluator:
+    """
+    Leverages an LLM to evaluate individual records.
+    """
+
+    def __init__(
+        self,
+        model: BaseEvalModel,
+        template: ClassificationTemplate,
+        name: str,
+        verbose: bool = False,
+    ) -> None:
+        """Initializer for LLMEvaluator.
+
+        Args:
+            model (BaseEvalModel): The LLM model to use for evaluation.
+            template (ClassificationTemplate): The evaluation template.
+            name (str): The name of the evaluator.
+            verbose (bool, optional): Whether to print verbose output.
+        """
+        self._model = model
+        self._template = template
+        self._name = name
+        self._verbose = verbose
+        self._parser = partial(_snap_to_rail, rails=template.rails, verbose=verbose)
```

Comment:
This doesn't seem to be all that necessary, and without it we could also use `dataclass(frozen=True)`. Ultimately, the parser should probably go live with the template instead.

Reply:
addressed in 53c9cca3